torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -176,19 +176,45 @@ template class UserRRef<py::object>;
 
 template <typename T>
 const T& OwnerRRef<T>::getValue() const {
-  // TODO: use callback to make this non-blocking
   std::unique_lock<std::mutex> lock(mutex_);
   valueCV_.wait(lock, [this] { return value_.has_value(); });
   return value_.value();
 }
 
+template <typename T>
+bool OwnerRRef<T>::hasValue() const {
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
`std::lock_guard` is supposedly cheaper if you don't have to unlock.

Never measured this or saw any proof though.

Reply:
ah right, I guess std::lock_guard can avoid a conditional in the destructor. I'll switch.